tests/org.eclipse.e4.ui.tests/src/org/eclipse/e4/ui/tests/workbench/PartRenderingEngineTests.java

```diff
@@ -826,17 +825,16 @@ public void testSelectedElementNullingParentChange() {
 		// the selected element doesn't change its value
 		container.setSelectedElement(partA);
 		container.getChildren().remove(partB);
-		assertTrue(
+		assertEquals(
 				"Changing the parent of a non-selected element should not change the value of the container's seletedElement",
-				container.getSelectedElement() == partA);
+				partA, container.getSelectedElement());
 
 		// Ensure that changing the parent of the selected element
 		// results in it going null
 		container.setSelectedElement(partA);
 		container.getChildren().remove(partA);
-		assertTrue(
-				"Changing the parent of the selected element should have set the field to null",
-				container.getSelectedElement() == null);
+		assertNotEquals("Changing the parent of the selected element should have set the field to null",
```

Comment:
This assert seems wrong, shouldn't it be assertNull?

Reply:
Sure.